### PR TITLE
Update finetuning scripts

### LINKED ISSUE #1009: Log learning rate and epoch in all scripts
Each script logs something else. We should log all basic metrics in all scripts the same way.

### PR DESCRIPTION
**Pull request by awaelchli**
Fixes some quirks and inconsistencies in the finetuning scripts.

- Log the same data in full.py and lora.py
- Lora's checkpoints had iter-prefix but saved on step
- Fix StopIteration when `max_iters > len(val_dataloader)`

Fixes #1009

### CODE REVIEW

File: finetune/adapter.py [hunk 1]
```diff
@@ -241,10 +241,10 @@ def validate(
 ) -> torch.Tensor:
     fabric.print("Validating ...")
     model.eval()
-    losses = torch.zeros(eval.max_iters)
-    val_iterator = iter(val_dataloader)
-    for k in range(eval.max_iters):
-        batch = next(val_iterator)
+    losses = torch.zeros(min(len(val_dataloader), eval.max_iters))
```
Comment (carmocca) [inline finetune/adapter.py:244]: Should you do this change for pretrain too?

File: finetune/lora.py [hunk 2]
```diff
@@ -238,25 +243,40 @@ def fit(
...
             t0 = time.perf_counter()
             val_loss = validate(fabric, model, val_dataloader, tokenizer, eval, data)
             t1 = time.perf_counter() - t0
             fabric.print(f"iter {iter_num}: val loss {val_loss.item():.4f}, val time: {t1 * 1000:.2f} ms")
+            metrics = {"val_loss": val_loss, "val_ppl": math.exp(val_loss)}
+            fabric.log_dict(metrics, step=iter_num)
             fabric.barrier()
         if not is_accumulating and step_count % train.save_interval == 0:
-            checkpoint_file = out_dir / f"iter-{iter_num:06d}" / "lit_model.pth"
+            checkpoint_file = out_dir / f"step-{step_count:06d}" / "lit_model.pth"
```
Comment (carmocca) [inline finetune/lora.py:279]: I don't see this change applied to the other files.

Comment (awaelchli) [reply finetune/lora.py:279]: I was hoping you wouldn't notice

**Comment by awaelchli** [comment]
> At some point we should side by side diff all the fine-tuning scripts to notice the differences and remove any that are not necessary

That's what I did in this PR. It's not removing all inconsistencies, but a reasonable chunk.
